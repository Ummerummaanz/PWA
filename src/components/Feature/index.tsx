import React from 'react';

const Feature: React.FC = () => (
  <div>{`#__FEATURE_NAME__#`}</div>
);

export default Feature;
