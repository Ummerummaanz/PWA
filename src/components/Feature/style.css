.feature-header-toolbar__root {
  display: flex;
  align-items: center;
}
