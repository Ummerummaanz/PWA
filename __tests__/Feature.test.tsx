import * as React from 'react';
import { render, waitFor } from '@testing-library/react';
import { getUserProfile } from 'oasis-feature-api';
import Feature from '../src/components/Feature';

const defaultAvatar = {
  fields: {
    title: 'avatar',
    file: {
      url: 'https://test/avatar',
    },
  },
};

const file = new File([''], 'avatar.png');
const apiAvatar = { hashCode: '1234', data: file };

jest.mock('oasis-os-contentful', () => ({
  useField: jest.fn((field: string) => {
    if (field === 'avatar') return defaultAvatar;
    if (field === 'greeting') return 'hello';
    return undefined;
  }),
}));

jest.mock('oasis-feature-api', () => ({
  getUserProfile: jest.fn(),
  getUserAvatar: jest.fn(() => {
    return Promise.resolve(apiAvatar);
  }),
}));

describe('Feature', () => {
  it.each([
    ['with Avatar', true],
    ['without Avatar', false],
  ])('should get welcome message %s', async (_, hasAvatarImage) => {
    (getUserProfile as jest.Mock).mockReturnValue(
      Promise.resolve({
        public: { firstName: 'test', hasAvatarImage },
      }),
    );
    const { container, getByTestId, getByAltText } = render(<Feature />);
    getByTestId('avatar__image');
    getByTestId('greeting_message');
    getByAltText('avatar');
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
