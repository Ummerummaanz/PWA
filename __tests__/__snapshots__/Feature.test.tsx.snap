// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature should get welcome message with Avatar 1`] = `
<div>
  <div
    class="feature-header-toolbar__root"
  >
    <div
      class="feature-header-toolbar__avatar"
    >
      <img
        alt="avatar"
        class="feature-header-toolbar__avatar__image"
        data-testid="avatar__image"
        src="https://test/avatar"
      />
    </div>
    <div
      class="feature-header-toolbar__greeting__message"
      data-testid="greeting_message"
    >
      Hello
       
      !
    </div>
    <div
      class="feature-header-toolbar__help"
    >
      <a
        data-testid="redirect__url"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="avatar"
          data-testid="help__image"
          src="https://test/avatar"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Feature should get welcome message with Upload 1`] = `
<div>
  <div
    class="feature-header-toolbar__root"
  >
    <div
      class="feature-header-toolbar__avatar"
    >
      <img
        alt="avatar"
        class="feature-header-toolbar__avatar__image"
        data-testid="avatar__image"
        src="https://test/avatar"
      />
    </div>
    <div
      class="feature-header-toolbar__greeting__message"
      data-testid="greeting_message"
    >
      Hello
       
      !
    </div>
    <div
      class="feature-header-toolbar__help"
    >
      <a
        data-testid="redirect__url"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="avatar"
          data-testid="help__image"
          src="https://test/avatar"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Feature should get welcome message without Avatar 1`] = `
<div>
  <div
    class="feature-header-toolbar__root"
  >
    <div
      class="feature-header-toolbar__avatar"
    >
      <img
        alt="avatar"
        class="feature-header-toolbar__avatar__image"
        data-testid="avatar__image"
        src="https://test/avatar"
      />
    </div>
    <div
      class="feature-header-toolbar__greeting__message"
      data-testid="greeting_message"
    >
      Hello
       
      !
    </div>
    <div
      class="feature-header-toolbar__help"
    >
      <a
        data-testid="redirect__url"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="avatar"
          data-testid="help__image"
          src="https://test/avatar"
        />
      </a>
    </div>
  </div>
</div>
`;
